subsys/net/buf.c

```diff
@@ -330,7 +330,9 @@ struct net_buf *net_buf_alloc_len(struct net_buf_pool *pool, size_t size,
 			return NULL;
 		}
 
+#if __ASSERT_ON
```

Comment:
IMHO it would be better to place this around `NET_BUF_ASSERT)` define, then all the assert calls would be covered.
So something like
```
#if __ASSERT_ON
#define NET_BUF_ASSERT(cond, ...) __ASSERT(cond, "" __VA_ARGS__)
#else 
#define NET_BUF_ASSERT(...)
#endif 
```

Reply:
@jukkar that's what I thought first as well, but the issue here is that there's a variable that's behind `#if __ASSERT_ON` so changing the `NET_BUF_ASSERT()` macro doesn't help.

Reply:
Ah, indeed, missed that one.